rpc/v10.ts

```diff
@@ -1 +1,2606 @@
+/**
+ * @unstable
+ */
+export interface RPCGetUserArgs {
+	/**
+	 * @unstable id of the user to get
+	 */
+	id: Snowflake;
+}
+
+/**
+ * https://discord.com/developers/docs/topics/rpc#getvoicesettings-get-voice-settings-response-structure
+ */
+export interface RPCGetVoiceSettingsResultData {
+	/**
+	 * input settings
+	 */
+	input: RPCVoiceSettingsInput;
+	/**
+	 * output settings
+	 */
+	output: RPCVoiceSettingsOutput;
+	/**
+	 * voice mode settings
+	 */
+	mode: RPCVoiceSettingsMode;
+	/**
+	 * state of automatic gain control
+	 */
+	automatic_gain_control: boolean;
+	/**
+	 * state of echo cancellation
+	 */
+	echo_cancellation: boolean;
+	/**
+	 * state of noise suppression
+	 */
+	noise_suppression: boolean;
+	/**
+	 * state of voice quality of service
+	 */
+	qos: boolean;
+	/**
+	 * state of silence warning notice
+	 */
+	silence_warning: boolean;
+	/**
+	 * state of self-deafen
+	 */
+	deaf: boolean;
+	/**
+	 * state of self-mute
+	 */
+	mute: boolean;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RPCGetVoiceSettingsArgs {}
+
+/**
+ * Returns the [Get Channel](https://discord.com/developers/docs/topics/rpc#getchannel) response, or `null` if none.
+ */
+export type RPCSelectTextChannelResultData = RPCGetChannelResultData | null;
+/**
+ * https://discord.com/developers/docs/topics/rpc#selecttextchannel-select-text-channel-argument-structure
+ */
+export interface RPCSelectTextChannelArgs {
+	/**
+	 * channel id to join (or `null` to leave)
+	 */
+	channel_id: Snowflake | null;
+	/**
+	 * asynchronously join channel with time to wait before timing out
+	 */
+	timeout?: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface RPCSetActivityResultData {}
+/**
+ * https://discord.com/developers/docs/topics/rpc#setactivity-set-activity-argument-structure
+ */
+export interface RPCSetActivityArgs {
+	/**
+	 * the application's process id
+	 */
+	pid: number;
+	/**
+	 * the rich presence to assign to the user
+	 */
+	activity?: Partial<
+		Omit<GatewayActivity, 'created_at' | 'id' | 'timestamps'> & Partial<Pick<GatewayActivity, 'timestamps'>>
+	>;
+}
+
+/**
+ * https://discord.com/developers/docs/topics/rpc#setvoicesettings-set-voice-settings-argument-and-response-structure
+ */
+export type RPCSetVoiceSettingsResultData = RPCGetVoiceSettingsResultData;
+/**
+ * https://discord.com/developers/docs/topics/rpc#setvoicesettings-set-voice-settings-argument-and-response-structure
+ *
+ * @note Discord only supports a single modifier of voice settings at a time over RPC. If an app changes voice settings, it will lock voice settings so that other apps connected simultaneously lose the ability to change voice settings. Settings reset to what they were before being changed after the controlling app disconnects. When an app that has previously set voice settings connects, the client will swap to that app's configured voice settings and lock voice settings again.
```

Comment:
```suggestion
 * @remarks Discord only supports a single modifier of voice settings at a time over RPC. If an app changes voice settings, it will lock voice settings so that other apps connected simultaneously lose the ability to change voice settings. Settings reset to what they were before being changed after the controlling app disconnects. When an app that has previously set voice settings connects, the client will swap to that app's configured voice settings and lock voice settings again.
```